pyvista/plotting/plotting.py

```diff
@@ -1959,17 +1962,19 @@ def add_volume(self, volume, scalars=None, clim=None, resolution=None,
         cmap = kwargs.pop('colormap', cmap)
         culling = kwargs.pop("backface_culling", culling)
 
-        # account for legacy behavior
-        if 'stitle' in kwargs:  # pragma: no cover
-            warnings.warn(USE_SCALAR_BAR_ARGS, PyvistaDeprecationWarning)
-            scalar_bar_args.setdefault('title', kwargs.pop('stitle'))
-
         if "scalar" in kwargs:
             raise TypeError("`scalar` is an invalid keyword argument for `add_mesh`. Perhaps you mean `scalars` with an s?")
         assert_empty_kwargs(**kwargs)
 
+        # Avoid mutating input
         if scalar_bar_args is None:
             scalar_bar_args = {}
```

Comment:
I know this is unrelated to the PR, but I'm wondering if it makes sense that `n_colors` is not set for `add_mesh`. It might not hurt to check that `n_colors` affects the colorbar for `add_mesh` .

Reply:
It would also be nice to extract out the common functionality for these things into a helper to make it consistent.  I think this should be a different PR as it is about functionality (with a possibly breaking change) rather than fixing an unintended effect.